### PR TITLE
SideNavigation: group `expanded` & `onExpand` prop added

### PR DESCRIPTION
**Pull request by AlbertCarreras**
SideNavigation: group `expanded` & `onExpand` prop added

Screenshot by Dropbox Capture
Screenshot by Dropbox Capture

### CODE REVIEW

File: docs/pages/web/sidenavigation.js [hunk 1]
```diff
@@ -383,8 +386,46 @@ To prevent visual overload, do not include counters in the parent if the childre
...
+            sandpackExample={<SandpackExample code={nestedExample} name="Nested example" />}
+          />
+        </MainSection.Subsection>
+
+        <MainSection.Subsection
+          title="Group display"
+          columns={2}
+          description="Group display can be \:
+1. static: group items are visible and there isn't expanding/collapsing behavior
+2. expandable: group items aren't visible un;ess one children is active and there is expanding/collapsing behavior
```
Comment (rlingineni) [inline docs/pages/web/sidenavigation.js:398]: typo

File: packages/gestalt/src/SideNavigation/GroupContent.js [hunk 2]
```diff
@@ -12,7 +12,7 @@ import Text from '../Text';
 import { type Indexable } from '../zIndex';
 
 type IconType = $Keys<typeof icons> | { __path: string };
-type Display = 'expandable' | 'static';
+type Display = 'expandable' | 'expanded' | 'static';
```
Comment (rlingineni) [inline packages/gestalt/src/SideNavigation/GroupContent.js:15]: Wouldn't it be cleaner if expanded was it's own prop? 

Like `<Sidenavigation.Group type="expandable"  expanded />`

Expandable vs Expanded seems a little confusing, and the states don't seem that different no?

Comment (rlingineni) [reply packages/gestalt/src/SideNavigation/GroupContent.js:15]: Expanded seems like a forced override essentially

**Review by rlingineni — APPROVED** [review]
r+ but comments on api

File: packages/gestalt/src/SideNavigation/GroupContent.js (hunk 2 above)
Comment (AlbertCarreras) [reply packages/gestalt/src/SideNavigation/GroupContent.js:15]: I agree w ```<Sidenavigation.Group type="expandable"  expanded />```

My only caveat is ```<Sidenavigation.Group type="static"  expanded />``` this now becomes possible.

Comment (rlingineni) [reply packages/gestalt/src/SideNavigation/GroupContent.js:15]: That's ok right? It's expanded anyways. It's a dead prop at that point

Comment (rlingineni) [reply packages/gestalt/src/SideNavigation/GroupContent.js:15]: Since all static displays are expanded by default

Comment (AlbertCarreras) [reply packages/gestalt/src/SideNavigation/GroupContent.js]: i think i prefer 

`'static' | 'expandable' | 'expandableExpanded'`


1. We avoid booleans
2. We avoid dead code
3. expandableExpanded is more explicit

I thought about expandable > expandableCollapsed but... that's a breaking change. but we could discuss.

Comment (rlingineni) [reply packages/gestalt/src/SideNavigation/GroupContent.js]: That seems confusing, are you sure?

It seems to me, just from looking at the examples, that expandable means -> collapse icons are showing vs they are not showing. 

expandableExpanded seems like a lot,

`Expanded` is a state variable. Essentially, we're making it a controlled component. It's like `checked`.  In static mode, there are no "expand" states, so it's not that important. Maybe there's a flow type to disable that case

Comment (AlbertCarreras) [reply packages/gestalt/src/SideNavigation/GroupContent.js]: we are not making it a controlled component, we are just setting a new initial state that controls if a group should be be initialized (collapsed vs expanded). But the user can still go and change that and it's still uncontrolled. 

The engineer shouldn't keep the state collapsed/expanded outside, it's uncontrolled to that respect.

Comment (AlbertCarreras) [reply packages/gestalt/src/SideNavigation/GroupContent.js]: Actually I see... some other improvement here.

Comment (AlbertCarreras) [reply packages/gestalt/src/SideNavigation/GroupContent.js]: I see your point now. With expanded prop we can control them. I thought just have them open from the beginning, but posterior changes don't persist. so there's no real differences between that and static.

Comment (AlbertCarreras) [reply packages/gestalt/src/SideNavigation/GroupContent.js]: it probably makes more sense to control it. There are many implementations to control that. Then we need on onExpand too there.

File: packages/gestalt/src/SideNavigationGroup.js [hunk 3]
```diff
@@ -78,9 +86,11 @@ export default function SideNavigationGroup({
   children,
   counter,
   display = 'expandable',
+  expanded: groupExpanded,
```
Comment (rlingineni) [inline packages/gestalt/src/SideNavigationGroup.js:89]: it's not clear groupExpanded is the prop. What if we called it `expandedProp`?

File: docs/examples/sidenavigation/displayExpanded.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+import { Box, DeviceTypeProvider, SideNavigation } from 'gestalt';
+
+export default function Example(): ReactNode {
+  const [expandedElements, setExpandedElements] = useState<$ReadOnlyArray<string>>([
+    'Christmas',
+    'Classic Christmas',
+  ]);
+
+  return (
+    <DeviceTypeProvider deviceType="mobile">
```
[the remark this answers is not in the document]
Comment (AlbertCarreras) [reply docs/examples/sidenavigation/displayExpanded.js:12]: this is a leftover from ttesting , good catch

**Pull request merged**